src/sentry/static/sentry/app/actionCreators/integrations.jsx

```diff
@@ -49,3 +50,100 @@ export function addIntegrationToProject(orgId, projectId, integration) {
     }
   );
 }
+
+/**
+ * Delete a respository
+ *
+ * @param {Object} client ApiClient
+ * @param {String} orgId Organization Slug
+ * @param {Number} repositoryId Repository ID
+ */
+export function deleteRepository(client, orgId, repositoryId) {
+  addLoadingMessage();
+  let promise = client.requestPromise(`/organizations/${orgId}/repos/${repositoryId}/`, {
+    method: 'DELETE',
+  });
+  promise.then(
```

Comment:
I intentionally didn't return the chain result here as I didn't want to have to throw an error from the catch callback.

Reply:
Might be good to leave this as a code comment.